Contrib/FreeWilson/freewilson.py

```diff
@@ -161,19 +174,28 @@ def __init__(self, smiles, rgroup, count, coefficient, idx=None):
         self.idx = idx                # descriptor index
         self.dummies = tuple([int(x) for x in sorted(dummypat.findall(smiles))])
 
-        # Assemble some additive properties 
+        # Assemble some additive properties
         try:
             # will fail on some structures
-            self.mw = Descriptors.MolWt(Chem.MolFromSmiles(smiles))
+            m = Chem.MolFromSmiles(smiles)
+            self.mw = Descriptors.MolWt(m)
+            self.hvyct = Descriptors.HeavyAtomCount(mol)
         except:
             # guess the MW if we can't sanitize
             table = Chem.GetPeriodicTable()
             try:
+                m = Chem.MolFromSmiles(smiles, sanitize=False)
                 self.mw = 0.
-                for atom in Chem.MolFromSmiles(smiles, sanitize=False).GetAtoms():
-                    self.mw += table.GetAtomicWeight(atom.GetAtomicNum())
+                self.hvyct = 0
+                for atom in m.GetAtoms():
+                    atomicnum = atom.GetAtomicNum()
+                    self.mw += table.GetAtomicWeight(atomicnum)
+                    if atomicnum > 1:
+                        self.hvyct += 1
+                self.hvyct = Descriptors.HeavyAtomCount(mol)
```

Comment:
you're calculating this twice and both are overly complicated
```suggestion
                self.hvyct = m.GetNumHeavyAtoms()
```